aquadoggo_cli/src/main.rs

```diff
@@ -25,7 +27,7 @@ struct Cli {
 
     /// URLs of remote nodes to replicate with.
     #[arg(short, long)]
-    remote_node_addresses: Vec<String>,
+    remote_node_addresses: Vec<Multiaddr>,
```

Comment:
By using `Multiaddr` and `PeerId` (instead of `String`) we get validation of the input automatically :) No need to try and `.parse()` later on.